receiver/k8sobjectsreceiver/receiver.go

```diff
@@ -49,22 +55,71 @@ func newReceiver(params receiver.CreateSettings, config *Config, consumer consum
 		return nil, err
 	}
 
-	return &k8sobjectsreceiver{
-		client:   client,
-		setting:  params,
-		consumer: consumer,
-		objects:  config.Objects,
-		obsrecv:  obsrecv,
-		mu:       sync.Mutex{},
-	}, nil
+	objReceiver := &k8sobjectsreceiver{
+		client:         client,
+		setting:        params,
+		logger:         params.Logger,
+		consumer:       consumer,
+		objects:        config.Objects,
+		obsrecv:        obsrecv,
+		mu:             sync.Mutex{},
+		leaderElection: config.LeaderElection,
+	}
+
+	if config.LeaderElection.Enabled {
+		objReceiver.leaderElectionClient, err = config.getClient()
+		if err != nil {
+			return nil, err
+		}
+
+		// use component "type-name" if resource lock name not set.
+		if objReceiver.leaderElection.LockName == "" {
+			objReceiver.leaderElection.LockName = getLeaderElectionLockName(params.ID)
+		}
```

Comment:
I don't think this is a valid fallback. This way, the leader election will be shared across different collector deployments, deamonsets or statefulsets. If you don't want to implement https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/24662/files#r1291903892 for now, I think we need to make the the lock name field required for the end user

Reply:
yes, `UID` of collector could be implemented by another separate followed-up PR